src/main/java/com/feedzai/commons/sql/abstraction/engine/impl/H2Translator.java

```diff
@@ -235,6 +252,38 @@ public String translate(Query q) {
         return q.isEnclosed() ? ("(" + finalQuery + ")") : finalQuery;
     }
 
+    @Override
+    public String translate(final UpdateFrom updateFrom) {
+        final Expression from = updateFrom.getFrom();
+
+        if (from == null) {
+            return translate((Update) updateFrom);
+        }
+
+        inject(from);
+
+        // This engine does not support UPDATE FROM.
+        // to workaround this, we do the following https://stackoverflow.com/a/44845278
+        final Update update = new Update(updateFrom.getTable());
+        for (final Expression column : updateFrom.getColumns()) {
+            final RepeatDelimiter eq = (RepeatDelimiter) column;
+            final Expression leftColumn = eq.getExpressions().get(0);
+            final Expression rightColumn = eq.getExpressions().get(1);
+
+            update.set(new RepeatDelimiter(eq.getDelimiter(),
+                                           ImmutableList.of(leftColumn,
+                                                            select(rightColumn)
+                                                                    .from(from)
+                                                                    .where(updateFrom.getWhere())
+                                                                    .enclose())));
+        }
+
+        inject(update);
+
+        return update.translate() + " WHERE EXISTS (SELECT * FROM "
+                + from.translate() + " WHERE " + updateFrom.getWhere().translate() + ");";
```

Comment:
the `;` in the end is not needed and actually should not be used in these JDCB SQL statements